### PR TITLE
Use page.evaluateHandle when we want to await on document promises in integration tests

### PR DESCRIPTION
**Pull request by calixteman**
For reference: https://github.com/mozilla/pdf.js/pull/17378#issuecomment-1842864939

### CODE REVIEW

File: test/integration/test_utils.mjs [hunk 1]
```diff
@@ -254,35 +265,35 @@ async function dragAndDropAnnotation(page, startX, startY, tX, tY) {
...
-    });
+  const handle = await createPromise(page, resolve => {
+    window.PDFViewerApplication.eventBus.on("textlayerrendered", resolve);
   });
+  return awaitPromise(handle);
 }
 
 async function scrollIntoView(page, selector) {
-  await page.evaluate(sel => {
+  const handle = await page.evaluateHandle(sel => {
```
[the remark this answers is not in the document]
Comment (OrKoN) [reply test/integration/test_utils.mjs]: That's right, assuming that scrollIntoView is awaited at the call sites. But the current version also work!

**Review by OrKoN — APPROVED** [review]

File: test/integration/test_utils.mjs (hunk 1 above)
[the remark this answers is not in the document]
Comment (calixteman) [reply test/integration/test_utils.mjs]: It's probably better to have one pattern: it avoids to wonder which one we should choose and avoids to make the bad choice.
That said, @OrKoN thank you for the `evaluatorHandle` trick.

**Review by Snuffleupagus — CHANGES_REQUESTED** [review]
There's a few cases of `fdescribe` left in the patch, which thus disables most tests. Please revert those and run all integration-tests, and I'll do a final review later today (assuming passing tests).

File: test/integration/test_utils.mjs [hunk 2]
```diff
@@ -54,6 +54,18 @@ function loadAndWait(filename, selector, zoom, pageSetup) {
   );
 }
 
+function createPromise(page, callback) {
+  return page.evaluateHandle(
+    // eslint-disable-next-line no-eval
+    cb => [new Promise(eval(`(${cb})`))],
```
[the remark this answers is not in the document]
Comment (calixteman) [reply test/integration/test_utils.mjs:60]: If you've a better idea, I take it. But for now, we can likely keep it without any concerns.

File: test/integration/test_utils.mjs [hunk 3]
```diff
@@ -100,23 +112,29 @@ function getSelectedEditors(page) {
...
+              resolve(true);
+            }, timeOut);
+          }),
+        ]),
+      ];
+    },
+    eventName,
+    timeout
+  );
+  const hasTimedout = await handle.evaluate(([p]) => p);
```
Comment (Snuffleupagus) [inline test/integration/test_utils.mjs:137]: Nit: For overall consistency, shouldn't this use the new helper function instead?
```suggestion
  const hasTimedout = await awaitPromise(handle);
```

**Review by Snuffleupagus — APPROVED** [review]
r=me, with one more question; thank you!

**Pull request merged**